### PR TITLE
tests: add redeployment tests for ephemeral and persistent templates

### PR DESCRIPTION
**Pull request by pkubatrh**

### CODE REVIEW

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by praiskup** [comment]
[test-openshift]

File: test/run-openshift [hunk 1]
```diff
@@ -17,6 +17,18 @@ test -n "${IMAGE_NAME-}" || false 'make sure $IMAGE_NAME is defined'
 test -n "${VERSION-}" || false 'make sure $VERSION is defined'
 test -n "${OS-}" || false 'make sure $OS is defined'
 
+function assert_cmd_fails() {
+  local result=""
+
+  eval "$@" &>/dev/null || result=1
+  if [ "$result" = "1" ]; then
+    echo " PASS"
+    return 0
```
Comment (praiskup) [inline test/run-openshift]: Slightly more readable version:
```bash
if eval "$@" &>/dev/null ; then
   echo " FAIL"
   return 1
fi
echo " PASS"
```

**Comment by praiskup** [comment]
looks fine, thanks

File: test/run-openshift (hunk 1 above)
Comment (pkubatrh) [reply test/run-openshift]: Added your version of assert_cmd_fails. Thanks for the suggestion!

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
Tests are green, merging. Thanks for review

**Pull request merged**